### PR TITLE
Akkaserverless discovery with static codegen from proto

### PR DESCRIPTION
**Pull request by andreaTP**
In this PR we switch to use `grpc-tools` instead of `protobuf-js` for static code generation instead of runtime loading for the Akkaserverless discovery service.

This shows how much better the static generation is, as opposed to dynamic proto loading:
 - no need for runtime generation
 - you can inspect the code generated
 - tooling works! (autocompletion actually works!)
 - easier setup
 
Byproduct:
 - moved away from `grpc`(which is deprecated) to `@grpc/grpc-js`
 - more tests!
 - `Akkaserverless.start` return a Promise (this is actually a little but breaking change)
 - fixed doc links from `js-services` -> `javascript`

### CODE REVIEW

File: sdk/package-lock.json [hunk 1]
```diff
@@ -1,7093 +1,8 @@
 {
   "name": "@lightbend/akkaserverless-javascript-sdk",
   "version": "0.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply sdk/package-lock.json]: Fixed! Thanks, we need to use NPM 7.
Should we document it?

File: sdk/src/akkaserverless.js [hunk 2]
```diff
@@ -174,126 +190,145 @@ class AkkaServerless {
...
+
+    return msg;
+  }
+
+  reportError(call, callback) {
+    const msg = this.reportErrorLogic(call.request)
     
     console.error(msg);
-    callback(null, {});
+    callback(null, new google_protobuf_empty_pb.Empty().toObject());
```
[the remark this answers is not in the document]
Comment (andreaTP) [reply sdk/src/akkaserverless.js:331]: well spotted, that's a leftover!

File: sdk/src/akkaserverless.js [hunk 3]
```diff
@@ -174,126 +190,145 @@ class AkkaServerless {
...
-      protocolMinorVersion: protocolVersion.minor
-    }
+    const serviceInfo = new discovery_messages.ServiceInfo();
+    serviceInfo.setServiceName(this.options.serviceName);
+    serviceInfo.setServiceVersion(this.options.serviceVersion);
+    serviceInfo.setServiceRuntime(process.title + " " + process.version);
+    serviceInfo.setSupportLibraryName(packageInfo.name);
+    serviceInfo.setSupportLibraryVersion(packageInfo.version);
+    serviceInfo.setProtocolMajorVersion(protocolVersion.major);
+    serviceInfo.setProtocolMinorVersion(protocolVersion.minor);
```
[the remark this answers is not in the document]
Comment (johanandren) [reply sdk/src/akkaserverless.js:247]: Yeah, it's fine I guess, just a bit chatty.

File: sdk/package-lock.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (pvlugter) [reply sdk/package-lock.json]: Yes, would be good to document it. This was a change that you actually originally made in #19.

It does mean that when `node `is installed (or probably also when updated to a newer 14.x version) than `npm` needs to be upgraded each time as well.

**Review by pvlugter — APPROVED** [review]
Switching to statically generated protobuf is useful.

And the other changes look good: updating to grpc-js (resolves https://github.com/lightbend/akkaserverless-framework/issues/601), updating the doc links, adding more tests.

**Pull request merged**